### PR TITLE
jak2: shadow renderer and blackout

### PR DESCRIPTION
**Pull request by Hat-Kid**
Still a WIP, shadows are a bit broken, use the Jak 1 VU1 code (Jak 2 VU1 breaks them even more) and there are some ugly hacks in the renderer

### CODE REVIEW

File: decompiler/config/jak2/ntsc_v1/type_casts.jsonc [hunk 1]
```diff
@@ -10237,5 +10245,131 @@
...
+    [34, "t1", "(pointer gs-reg)"],
+    [38, "t1", "(pointer gs-reg)"],
+    [42, "t1", "(pointer gs-reg)"],
+    [36, "t1", "(pointer gs-zbuf)"],
+    [40, "t1", "(pointer gs-test)"],
+    [44, "t1", "(pointer gs-alpha)"],
+    [45, "t1", "(pointer gs-reg)"],
+    [47, "t1", "(pointer gs-reg64)"],
+    [49, "t1", "(pointer gs-reg)"],
+    [51, "t1", "(pointer gs-rgbaq)"],
```
Comment (ManDude) [inline decompiler/config/jak2/ntsc_v1/type_casts.jsonc:10278]: Can probably get rid of these and trigger a `dma-buffer-add-gs-set` to decompile.

File: decompiler/config/jak2/ntsc_v1/type_casts.jsonc [hunk 2]
```diff
@@ -10237,5 +10245,131 @@
...
+    [73, "t4", "(pointer gs-reg64)"],
+    [81, "t4", "(pointer gs-reg)"],
+    [84, "t4", "(pointer gs-reg)"],
+    [62, "t4", "(pointer gs-zbuf)"],
+    [71, "t4", "(pointer gs-xy-offset)"],
+    [79, "t4", "(pointer gs-tex0)"],
+    [82, "t4", "(pointer gs-tex1)"],
+    [85, "t4", "(pointer gs-miptbp)"],
+    [87, "t4", "(pointer gs-reg)"],
+    [96, "t4", "(pointer gs-clamp)"],
```
Comment (ManDude) [inline decompiler/config/jak2/ntsc_v1/type_casts.jsonc:10306]: Can probably get rid of these and trigger a `dma-buffer-add-gs-set` to decompile.

File: decompiler/config/jak2/ntsc_v1/type_casts.jsonc [hunk 3]
```diff
@@ -10237,5 +10245,131 @@
...
+    [181, "a3", "(pointer gs-zbuf)"],
+    [188, "a3", "(pointer gs-frame)"],
+    [191, "a3", "(pointer uint64)"],
+    [206, "a3", "(pointer gs-reg64)"],
+    [213, "a3", "(pointer gs-reg64)"],
+    [214, "a3", "(pointer uint64)"],
+    [216, "a3", "(pointer gs-reg64)"],
+    [211, "a3", "(pointer gs-frame)"],
+    [204, "a3", "(pointer gs-zbuf)"],
+    [200, "a3", "(pointer gs-test)"],
```
Comment (ManDude) [inline decompiler/config/jak2/ntsc_v1/type_casts.jsonc:10329]: Can probably get rid of these and trigger a `dma-buffer-add-gs-set` to decompile.

File: decompiler/config/jak2/ntsc_v1/type_casts.jsonc [hunk 4]
```diff
@@ -10237,5 +10245,131 @@
...
+    [69, "a0", "(pointer uint64)"],
+    [71, "a0", "(pointer gs-reg64)"],
+    [75, "a0", "(pointer gs-reg64)"],
+    [82, "a0", "(pointer gs-reg64)"],
+    [83, "a0", "(pointer uint64)"],
+    [85, "a0", "(pointer gs-reg64)"],
+    [86, "a0", "(pointer uint64)"],
+    [88, "a0", "(pointer gs-reg64)"],
+    [73, "a0", "(pointer gs-test)"],
+    [80, "a0", "(pointer gs-frame)"],
```
Comment (ManDude) [inline decompiler/config/jak2/ntsc_v1/type_casts.jsonc:10353]: Same here.

**Comment by Hat-Kid** [comment]
Merged as part of #2632.

**Pull request closed without merging**